### PR TITLE
Fixes #20641 - clean the upgrade storage on upgrade finish

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

**Comment by iNecas** [comment]
Tests updated

**Comment by iNecas** [comment]
@swapab mind review?

File: lib/foreman_maintain/upgrade_runner.rb [hunk 1]
```diff
@@ -67,6 +67,9 @@ def run
           run_phase(phase)
         end
       end
+      unless quit?
```
Comment (swapab) [inline lib/foreman_maintain/upgrade_runner.rb]: Just a silly question. Any reason to prefer block check over an inline check?

**Comment by swapab** [comment]
Upgrade run works like a charm.!!! @iNecas  :trophy: 
Tested with following command:
```
./bin/foreman-maintain upgrade run --target-version 6.2.z --whitelist="check-old-foreman-tasks,disk-io,packages-update"
```
http://pastebin.test.redhat.com/511032 (Ignore the errors. Those are system specific)

---

Cleaning also works as expected.

Command:
```
./bin/foreman-maintain upgrade run --target-version 6.2.z --whitelist="check-old-foreman-tasks,disk-io,packages-update,installer-upgrade"
```
Stripped result:
```
Running procedures after migrating to Satellite 6.2.z
================================================================================
turn off maintenance mode:                                            [OK]
--------------------------------------------------------------------------------
re-enable sync plans: 
/ Total 0 sync plans are now enabled.                                 [OK]      
--------------------------------------------------------------------------------


Running checks after upgrading to Satellite 6.2.z
================================================================================
check for paused tasks:                                               [OK]
--------------------------------------------------------------------------------
Check whether system is self-registered or not: 
System is not self registered                                         [OK]
--------------------------------------------------------------------------------


--------------------------------------------------------------------------------
Upgrade finished.
```

---

Ready to merge. :white_check_mark:

**Comment by iNecas** [comment]
Thanks @swapab

File: lib/foreman_maintain/upgrade_runner.rb (hunk 1 above)
Comment (iNecas) [inline lib/foreman_maintain/upgrade_runner.rb]: just a personal preference. Sometimes the modifier statement can get overlooked. Probably not this case though

**Pull request merged**